### ISSUE TITLE
Guide: Chris Allen

### ISSUE THREAD

**Issue by bitemyapp** [lambdaconf/winter-retreat-2017#8]
I'm Chris Allen, an author of Haskell Programming from First Principles, commonly known as the [haskell book](http://haskellbook.com). I'm an uneducated working programmer that got tired of unnecessary trade-offs between types and expressiveness.

I originally started teaching so that I could spin up coworkers on Haskell, but found teaching itself to be more energizing than writing code.

Past topics I've covered in some way include:
- [lambda calculus, expressions, types, polymorphism, typeclasses, higher kinded types, functor/applicative/monad, foldable/traversable, reader, state, monad transformers, benchmarking and profiling, laziness, IO, and more](http://haskellbook.com)
- [Yesod and Persistent in practice](https://www.youtube.com/watch?v=uYXX1t3GrsE) (I use these at work)
- Rank N Types (I gave a talk on this at LC 16 and really enjoyed it)
- [Instance-local functional dependencies](https://github.com/bitemyapp/buttress/blob/master/src/Buttress/Time.hs)
- Practical applications of type families in Haskell (mono-traversable, Persistent, and stripe-haskell come to mind)

Topics I don't have much practice teaching with but would be happy to explore with someone include:
- Servant
- GHCJS (e.g. https://github.com/bitemyapp/ghcjs-starter-project )
- Parsing/tagging parts of speech (Such as with [Chatter](https://www.stackage.org/package/chatter))

I am excited to be a guide at the LambdaConf Winter Retreat! I'm not stuck on anything in this list, so if you have requests, please tell me as soon as possible!

Edit:

### Final talk description

Hello everyone! I'm known primarily for the book [Haskell Programming from First Principles](http://haskellbook.com) and for some free Haskell tutorials and guides. The material of my presentation at the winter retreat will be based on my upcoming book named [Haskell Almanac](http://lorepub.com/product/cookbook). The content will be focused on intermediate topics which are commonly needed in industry software projects. Think "web applications", except I may not devote much actual time to a web framework, but rather, the peripheral dependencies of a Haskell web application.

It is safe to assume that you should have the basics of Haskell down to get the full value of this lesson! The first 2/3s or 3/4s of the [Haskell Book](http://haskellbook.com) is a good baseline for what you should know. Remember that merely reading the material isn't sufficient, you will need to have done the exercises. If you're already using Haskell productively and can sling things like monad transformers without too much comfort, you probably do not need the book although some advanced readers have enjoyed the insights it offers.

**Comment by bitemyapp** [comment]
Requests so far have been:
- Web apps
- streaming libraries (machines, conduit, pipes)
- event sourcing
- `Data.ByteString.Lazy` and `Builder`
- `monad-control`

**Comment by mudphone** [comment]
I'd also be interested in learning about building web apps. Doing so with Yesod and Persistent, or GHCJS sounds fun!

**Comment by EncodePanda** [comment]
Yeah, Chris :) !! For me you can talk about anything, I'm all ears :)

**Comment by haroldcarr** [comment]
Interested in (in listed order):
- Practical applications of type families in Haskell
- Rank N Types
- Instance-local functional dependencies

**Comment by jdegoes** [comment]
My vote: Workshop on using both type families and functional dependencies, and showing how different scenarios benefit from either.

**Comment by serras** [comment]
+1 to @jdegoes 
I would love hearing about the interplay of type classes, open and closed type and data families, associated types, functional dependencies and injectivity annotations. They all look similar, but each of them is tricky in its own way.

**Comment by rezalesmana** [comment]
web apps with event sourcing would be great

**Comment by purefunctions** [comment]
GHCjs, and functional front-end in general

**Comment by ErrorNullPointer** [comment]
+1 to @haroldcarr 

I second his vote.

**Comment by astronuge** [comment]
+1 to lambda calculus, expressions, types, polymorphism, typeclasses, higher kinded types, functor/applicative/monad, foldable/traversable, reader, state, monad transformers, benchmarking and profiling, laziness, IO, and more

**Comment by bitemyapp** [comment]
For anyone wanting rank n types: https://www.youtube.com/watch?v=k0cZe0LVFI4

**Comment by purefunctions** [comment]
@bitemyapp Do you have the slides/code from the talk above? The screen isn't that observable in the video

**Comment by bitemyapp** [comment]
@emptyslot 

https://github.com/bitemyapp/presentations/blob/master/rank_n_types_talk/rankn.pdf

https://github.com/bitemyapp/presentations/blob/master/rank_n_types_talk/tuple.hs

https://github.com/bitemyapp/presentations/blob/master/rank_n_types_talk/ranks.hs

**Comment by gilligan** [comment]
> Servant

Personally I would be very interested in developing some small servant project. I would possibly consider the fact that you don't know Servant that well as a bonus because that way I could see how you explore new Haskell libraries/frameworks.

So developing a small servant app ideally with some tiny frontend part would be great for me. Maybe not thrilling as some of the fancy topics mentioned here but anyway.. ;)

**Comment by alexi21** [comment]
My work is all web applications so anything relating to that would be great.
